### DISCUSSION TITLE
How to refresh a controller with different data?

### DISCUSSION THREAD

**Discussion by ComputelessComputer** [jonataslaw/getx#1109] [Q&A]
I am creating a flutter mobile app using Firebase for backend solution.

I have a page where the user will be able to play a quiz and it takes the user back to home when it is over.

However the quizzes have different types and I am rendering them based on the type.

If I were to register an user answer response and save that to the DB, how should I reload the next quiz on the same page?

I have an idea to achieve this by rendering different views by listening to the change in the above document. But i haven't tried it yet. Would this be a "right" approach?

 ![simple diagram](https://user-images.githubusercontent.com/63365510/108860453-8d3d1e00-7631-11eb-8b4c-d3efaabf36ec.jpg)

**Comment by fer-ri** [comment]
There's nothing wrong or right, It depends on your requirement.

Just make sure you store the quiz as `Rx` and Flutter can detect its changes.

Usually, I use `equatable` on the Quiz model class, so Flutter can check the difference between the quiz.
